src/hb-ot-map.cc

```diff
@@ -390,5 +390,27 @@ hb_ot_map_builder_t::compile (hb_ot_map_t                  &m,
   }
 }
 
+unsigned int hb_ot_map_t::get_features (unsigned int start_offset, unsigned int *feature_count, hb_feature_t *out_features) const
+{
+  if (feature_count)
+  {
+    auto sub_features = features.as_array ().sub_array (start_offset, feature_count);
+
+    if (out_features)
+    {
+      for (unsigned int i = 0; i < sub_features.length; i++)
+      {
+        out_features[i].tag = sub_features[i].tag;
+        // FIXME: I should be using the mask to get the proper value, start, and
+        // end, but I'm not sure how to do that.
+        out_features[i].value = 1;
```

Comment:
Maybe we use value=0 if it's a partial feature, and 1 if it's global? Just check if it uses the global mask or not, for now?

Reply:
Or we can set it to *max* possible set value.

Reply:
lemme think a bit. A rather unintuitively, but:
- value set to *0* if a global feature,
- value set to max possible value for the shape plan. for binary non-global features this will turn to on, suggesting that the UI can expose a on/off for it. You just use the shift value here.

Reply:
Thinking about it more, may be it should just return feature tags. Global vs partial features would be interesting only for user features, but the caller can already track that. I’m leaning to renaming it `hb_ot_shape_plan_get_feature_tags()`. If we need the finer details in the future, we can have another API that returns `hb_feature_t`.

Reply:
Well that would possibly give the UI's the wrong impression that eg. they should expose `init` etc to the user. I'm fine with doing what you suggest for now.

Reply:
It depends on the intended use of the UI, an application like FontForge or FontGoggles will want to show all features anyway. A typesetting application should not show any required feature (i.e. hide rlig but show liga), but the API here won’t help with that. May be we should have an API to list required feature, but it would be a hard-coded list based on OpenType feature registry, so HarfBuzz might not be the best place for it as HarfBuzz does not itself make a distinction between required and not required default features.

Reply:
Right... Okay. Let's proceed then.

Reply:
Waiting for @iorsh to try the API and see if it works for his FontForge code.